src/snippets/responsive-image.liquid

```diff
@@ -0,0 +1,72 @@
+{%- comment -%}
+    It creates a style tag and it restricts an image from growing larger than its max resolution.
+
+    Dependencies:
+    - Lazysizes plugin (https://github.com/aFarkas/lazysizes) which enable responsive image with faster load time and better performance.
+    - Lazysizes Responsive Images as a Service (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/rias) To load the correct image size with our CDN
+    - Lazysizes Bgset (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/bgset) To use responsive images on background-image (CSS)
+
+    Accepts:
+    - image: {Object} Image object
+    - max_width: {Number} Max width of the image container
+    - max_height: {Number} Max height of the image container
+    - img_class: {String} class name of the <img />
+    - wrapper_class: {String} class name of the <div> wrapper
+
+    Usage:
+    In your liquid template file, copy the following line
+    - {% include 'responsive-image' with image: featured_image, img_class: "css-class", wrapper_class: "wrapper-css-class", max_width: 700, max_height: 800 %}
+{%- endcomment -%}
+
+{%- comment -%} Added incremental number to avoid conflict styling code when the same images are using this snippet {%- endcomment -%}
+{%- capture responsive_image_counter %}{% increment image_counter %}{% endcapture -%}
```

Comment:
Did you mean to do `{% increment responsive_image_counter %}` even though your implementation works here? Why this still works I'm not too sure, but I assume Liquid is assigning an undefined variable to nil and then type casting it to a 0 integer due to the use of `increment`.

Reply:
@chrisberthe Yes, you're right. When using `increment` tags, Liquid automatically creates a variable and default to `0`. In my case, even though I could named the `increment` variable anything, I'll change the name so it would be less confusing.